wasi-sdk.cmake

```
@@ -3,6 +3,10 @@
 # This is arbitrary, AFAIK, for now.
 cmake_minimum_required(VERSION 3.4.0)
 
+# Until Platform/WASI.cmake is upstream we need to inject the path to it
+# into CMAKE_MODULE_PATH.
+list(APPEND CMAKE_MODULE_PATH "${WASI_SDK_PREFIX}/cmake")
```

I am using the wasi-sdk release tarball, which has `share/cmake/wasi-sdk.cmake` and does not contain the platform. Should the platform file be included at `share/cmake/Platform/WASI.cmake`, the following change would be needed:

```suggestion
list(APPEND CMAKE_MODULE_PATH "${WASI_SDK_PREFIX}/share/cmake")
```

Doing these manually locally does seem to make it work.

---

@sbc100 Does this change make sense?

---

We should probably add some kind of minimal cmake test to the CI here too.

---

Just to confirm, the release tarball (for 12.0 at least) does not contain the platform file. I placed it to `SDK/share/cmake/Platform/WASI.cmake` to mirror directory layout here.

---

As @sbc100 mentioned, is there a way we can add a minimal test here to the CI to ensure that this keeps working?